### PR TITLE
508 add redux to qatomic

### PR DESCRIPTION
**Pull request by carlopascual**
This change is [Reviewable](https://reviewable.io/reviews/quadric/qatomic/4)

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (zvictor) [inline .vscode/settings.json]: noooo! 😨

File: generators/app/index.js [hunk 2]
```diff
@@ -4,9 +4,36 @@ var yeoman = require('yeoman-generator');
...
+  const componentKey = regex.exec(path);
+
+  return componentKey && componentKey[1];
+}
+
 module.exports = yeoman.Base.extend({
+
   prompting: function () {
     // Have Yeoman greet the user.
+    console.log(this.destinationRoot());
```
Comment (zvictor) [inline generators/app/index.js]: > +    console.log(this.destinationRoot());
> +    console.log(getComponentKey(this.destinationRoot()));

forgot to remove the loggers?

File: generators/app/templates/actionCreators.ejs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// import _ from 'lodash';
+import { createAction } from 'redux-actions';
+import { COMPONENT_KEY } from './constants';
+
+export const mySweetAction = createAction(`${COMPONENT_KEY}/DO_WHATEVER`, async () => {
+  return await doWhatever();
+});
```
Comment (zvictor) [inline generators/app/templates/actionCreators.ejs]: missing line break

File: generators/app/templates/actions.ejs [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import contextualize from '/imports/client/contextualize';
+import * as actions from './actionCreators';
+
+module.exports = contextualize(actions);
```
Comment (zvictor) [inline generators/app/templates/actions.ejs]: missing line break

File: generators/app/templates/reducer.ejs [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash';
```
Comment (zvictor) [inline generators/app/templates/reducer.ejs]: > +import _ from 'lodash';

please leave it commented

File: generators/app/templates/reducer.ejs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        isPending: false,
+      }),
+      REJECTED: (state, { payload }) => ({
+        ...state,
+        error: payload,
+        isPending: false,
+      }),
+    },
+  }, initial),
+};
```
Comment (zvictor) [inline generators/app/templates/reducer.ejs]: missing line break

**Review by zvictor — CHANGES_REQUESTED** [review]

**Comment by carlopascual** [comment]
Reviewed 8 of 8 files at r2.
Review status: 7 of 8 files reviewed at latest revision, 6 unresolved discussions.

---

*[generators/app/templates/actionCreators.ejs, line 7 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0le-r1-7:-KesQL9hNFe4e1OTH7TJ:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/generators/app/templates/actionCreators.ejs#L7)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

missing line break
</blockquote></details>

Done.

---

*[generators/app/templates/actions.ejs, line 4 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0lf-r1-4:-KesQMC9b9DouI0oYr9J:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/generators/app/templates/actions.ejs#L4)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

missing line break
</blockquote></details>

Done.

---

*[generators/app/templates/reducer.ejs, line 1 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0lh-r1-1:-KesQMpPkDGVXotbFHeo:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/generators/app/templates/reducer.ejs#L1)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

> +import _ from 'lodash';

please leave it commented
</blockquote></details>

Done.

---

*[.vscode/settings.json, line 1 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0ld-r1-1:-KesPDlvvpoxE5VybF-v:bl7v0we) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/.vscode/settings.json#L1)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

noooo! 😨
</blockquote></details>

:laughing: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Review status: 7 of 8 files reviewed at latest revision, 6 unresolved discussions.

---

*[generators/app/index.js, line 34 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0lc-r1-34:-KesQaz-nI1Gp87NKAC6:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/generators/app/index.js#L34)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

> +    console.log(this.destinationRoot());
> +    console.log(getComponentKey(this.destinationRoot()));

forgot to remove the loggers?
</blockquote></details>

Done.

---

*[generators/app/templates/reducer.ejs, line 29 at r1](https://reviewable.io:443/reviews/quadric/qatomic/4#-KesNVoZ0gtdwMhmO0lh-r1-29:-KesQ_vGug-V7CEPs3Ch:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/66517ed897e298bb6c54346141b42f4ef46c6bcd/generators/app/templates/reducer.ejs#L29)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

missing line break
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 6 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by zvictor** [comment]
Reviewed 1 of 6 files at r1, 4 of 8 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[generators/app/.DS_Store, line 0 at r2](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfB-QDpg_S_-IysU0wA:-KfB-QDpg_S_-IysU0wB:b-ib4774) ([raw file](https://github.com/quadric/qatomic/blob/0867f8ca6098599bb14d0e49b6f7a1322f3a2a3c/generators/app/.DS_Store#L0)):*
we should not push `.DS_Store` to the repo. Please add it to the `.gitignore` file.

---

*[generators/app/index.js, line 13 at r2](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfB0hH31ETCGT5CXXcF:-KfB0hH31ETCGT5CXXcG:bm1nuqe) ([raw file](https://github.com/quadric/qatomic/blob/0867f8ca6098599bb14d0e49b6f7a1322f3a2a3c/generators/app/index.js#L13)):*
> ```JavaScript
> * ex: blah/components/test/asdf => test/asdf
> *     blah/components/components/asdfasdf/asdfasdf => asdfasdf/asdfasdf
> *
> ```

good docs :)

---

*[generators/app/index.js, line 70 at r2](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfB0zkHSaVz3drVWNQF:-KfB0zkHSaVz3drVWNQG:b2f9tek) ([raw file](https://github.com/quadric/qatomic/blob/0867f8ca6098599bb14d0e49b6f7a1322f3a2a3c/generators/app/index.js#L70)):*
> props.componentKey = `${getComponentKey(this.destinationRoot())}/${props.name}` || props.name;

does it work well also for when there is no `props.name`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/.DS_Store, line  at r2](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfB-QDpg_S_-IysU0wA:-KfGPkN8evZE2_NVWqLK:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/0867f8ca6098599bb14d0e49b6f7a1322f3a2a3c/generators/app/.DS_Store#L)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

we should not push `.DS_Store` to the repo. Please add it to the `.gitignore` file.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 3 files at r3.
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 70 at r2](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfB0zkHSaVz3drVWNQF:-KfGhpPoSWDqS3kxAoEu:b-am0p07) ([raw file](https://github.com/quadric/qatomic/blob/0867f8ca6098599bb14d0e49b6f7a1322f3a2a3c/generators/app/index.js#L70)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

> props.componentKey = `${getComponentKey(this.destinationRoot())}/${props.name}` || props.name;

does it work well also for when there is no `props.name`?
</blockquote></details>

Just checked. It works now that I changed the implementation of the current folder (this.appname wasn't giving us the name of the current folder)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by zvictor** [comment]
Reviewed 1 of 3 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 40 at r4](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHPZHxVrzkjVQyhyTU:-KfHPZHxVrzkjVQyhyTV:b-pfjb38) ([raw file](https://github.com/quadric/qatomic/blob/2ff591807ddd9c2c7190c227ba4469baf8882cfe/generators/app/index.js#L40)):*
> ```JavaScript
>       'Welcome to the hunky-dory ' + chalk.red('Qatomic') + ' generator!'
>     ));
> ```

`const currentFolderName = this.destinationRoot().split('/').pop();`

`currentFolderName` should be safer than `currentFolder`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 40 at r4](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHPZHxVrzkjVQyhyTU:-KfHQZbcz0l-0AxHpcX0:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/2ff591807ddd9c2c7190c227ba4469baf8882cfe/generators/app/index.js#L40)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

`const currentFolderName = this.destinationRoot().split('/').pop();`

`currentFolderName` should be safer than `currentFolder`
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by zvictor** [comment]
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 40 at r4](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHPZHxVrzkjVQyhyTU:-KfHRVhTyqn09WZgtiCt:b3kre) ([raw file](https://github.com/quadric/qatomic/blob/2ff591807ddd9c2c7190c227ba4469baf8882cfe/generators/app/index.js#L40)):*
<details><summary><i>Previously, carlopascual (curiosity) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 40 at r4](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHPZHxVrzkjVQyhyTU:-KfHTvoB06OtWqqB4k5G:bk94svv) ([raw file](https://github.com/quadric/qatomic/blob/2ff591807ddd9c2c7190c227ba4469baf8882cfe/generators/app/index.js#L40)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

?
</blockquote></details>

`const currentFolderName = this.destinationRoot().split('/').pop();` only works inside the functions in `yeoman.Base.extend(..`.. Will declare multiple consts then inside these functions

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Review status: 4 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[generators/app/index.js, line 40 at r4](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHPZHxVrzkjVQyhyTU:-KfHVCub3kiPrlzeSlSg:b-896fix) ([raw file](https://github.com/quadric/qatomic/blob/2ff591807ddd9c2c7190c227ba4469baf8882cfe/generators/app/index.js#L40)):*
<details><summary><i>Previously, carlopascual (curiosity) wrote…</i></summary><blockquote>

`const currentFolderName = this.destinationRoot().split('/').pop();` only works inside the functions in `yeoman.Base.extend(..`.. Will declare multiple consts then inside these functions
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by zvictor** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[generators/app/index.js, line 33 at r6](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHWl7iil9nsdHav-bp:-KfHWl7iil9nsdHav-bq:bvwutq7) ([raw file](https://github.com/quadric/qatomic/blob/6825fb6ee12d47b7005ff3d95d143ce80eefe6d1/generators/app/index.js#L33)):*
> ```JavaScript
>   prompting: function () {
>     // Have Yeoman greet the user.
>     const currentFolderName = this.destinationRoot().split('/').pop();
> ```

one way to avoid repeating code would setting `currentFolderName` as an attr. of the instance:
`this.currentFolderName = this.destinationRoot().split('/').pop();`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[generators/app/index.js, line 33 at r6](https://reviewable.io:443/reviews/quadric/qatomic/4#-KfHWl7iil9nsdHav-bp:-KfHlfdVveuy11p0ukG-:b-bxjxyj) ([raw file](https://github.com/quadric/qatomic/blob/6825fb6ee12d47b7005ff3d95d143ce80eefe6d1/generators/app/index.js#L33)):*
<details><summary><i>Previously, zvictor (zVictor) wrote…</i></summary><blockquote>

one way to avoid repeating code would setting `currentFolderName` as an attr. of the instance:
`this.currentFolderName = this.destinationRoot().split('/').pop();`
</blockquote></details>

converted to `this.currentFolderName` @ r7

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by carlopascual** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Comment by zvictor** [comment]
Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/quadric/qatomic/4)*

**Pull request merged**